pkg/controller/backup_storage_location_controller.go

```diff
@@ -46,6 +47,97 @@ const (
 	bslValidationEnqueuePeriod = 10 * time.Second
 )
 
+// sanitizeStorageError cleans up verbose HTTP responses from cloud provider errors,
+// particularly Azure which includes full HTTP response details and XML in error messages.
+// It extracts the error code and message while removing HTTP headers and response bodies.
+func sanitizeStorageError(err error) string {
```

Comment:
can we also enhance this function to use storage sas regex and remove secrets from being shown in status/ logging.

Reply:
Done